Tests/tests/test_SystemAudioPlayer.cpp

```diff
@@ -0,0 +1,1356 @@
+         response
+    ));
+
+    size_t idPos = response.find("\"id\"");
+
+    if (idPos != string::npos) {
+
+        size_t idStart = response.find(':', idPos) + 1;
+        size_t idEnd = response.find(',', idPos);
+
+        std::string idSubstring = response.substr(idStart, idEnd - idStart);
+
+        int speechId = std::stoi(idSubstring);
+
+        std::cout << "Speech ID: " << speechId << std::endl;
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("resume"), _T("{\"id\": ") + std::to_string(speechId) + _T("}"), response));
+
+        EXPECT_EQ(response, _T("{\"success\":false}"));
+    } else {
+        EXPECT_TRUE(false) << "Error: 'id' not found in the response.";
+    }
+}
+
+/**
+ * @name  : SAPResumeWithOutOpen
+ * @brief : Resume the invalid player
+ * 
+ * @param[in]   :  id
+ * @return      :  {success: false}
+ */
+
+TEST_F(SAPInitializedTest,SAPResumeWithOutOpen) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("resume"), _T("{\"id\": 1}"), response));
+    EXPECT_EQ(response, _T("{\"success\":false}"));
+}
+
+/*******************************************************************************************************************
+ * Test function for pause
+ * Pause                   :
+ *                Closes the specified Player
+ *
+ *                @return Response object success status
+ * Use case coverage:
+ *                @Success : 1
+ *                @Failure : 1
+ ********************************************************************************************************************/
+
+/**
+ * @name  : SAPPause
+ * @brief : Pause the player instance
+ * 
+ * @param[in]   :  id
+ * @return      :  {success: false}
+ */
+
+TEST_F(SAPInitializedTest, SAPPause) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection,
+        _T("open"), 
+        _T("{\"audiotype\": \"pcm\",\"sourcetype\": \"websocket\",\"playmode\": \"system\" }"),
+         response
+    ));
+
+    size_t idPos = response.find("\"id\"");
+
+    if (idPos != string::npos) {
+
+        size_t idStart = response.find(':', idPos) + 1;
+        size_t idEnd = response.find(',', idPos);
+
+        std::string idSubstring = response.substr(idStart, idEnd - idStart);
+
+        int speechId = std::stoi(idSubstring);
+
+        std::cout << "Speech ID: " << speechId << std::endl;
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("pause"), _T("{\"id\": ") + std::to_string(speechId) + _T("}"), response));
+
+        EXPECT_EQ(response, _T("{\"success\":false}"));
+    } else {
+        EXPECT_TRUE(false) << "Error: 'id' not found in the response.";
+    }
+}
+
+/**
+ * @name  : SAPPauseWithOutOpen
+ * @brief : Pause the invalid player
+ * 
+ * @param[in]   :  id
+ * @return      :  {success: false}
+ */
+TEST_F(SAPInitializedTest,SAPPauseWithoutOpen) {
```

Comment:
Is the test supposed to be 'SAPPauseWithoutPlay'?

Reply:
done